apps/astarte_trigger_engine/lib/astarte_trigger_engine/events_consumer.ex

```diff
@@ -62,41 +62,65 @@ defmodule Astarte.TriggerEngine.EventsConsumer do
     handle_simple_event(realm, device_id, headers, event_type, event, timestamp_ms)
   end
 
-  defp handle_simple_event(realm, device_id, headers_map, event_type, event, timestamp_ms) do
+  defp handle_simple_event(
+         realm,
+         device_id,
+         headers_map,
+         event_type,
+         event,
+         timestamp_ms
+       ) do
     with {:ok, trigger_id} <- Map.fetch(headers_map, "x_astarte_parent_trigger_id"),
-         {:ok, action} <- retrieve_trigger_configuration(realm, trigger_id),
-         {:ok, payload} <-
-           event_to_payload(realm, device_id, event_type, event, action, timestamp_ms),
-         {:ok, headers} <- event_to_headers(realm, device_id, event_type, event, action),
-         :ok <- execute_action(payload, headers, action) do
-      :telemetry.execute(
-        [:astarte, :trigger_engine, :http_action_executed],
-        %{},
-        %{realm: realm, status: :ok}
-      )
+         {:ok, action} <- retrieve_trigger_configuration(realm, trigger_id) do
+      with {:ok, payload} <-
+             event_to_payload(realm, device_id, event_type, event, action, timestamp_ms),
+           {:ok, headers} <- event_to_headers(realm, device_id, event_type, event, action) do
+        payload_size = payload |> :erlang.term_to_binary() |> :erlang.byte_size()
```

Comment:
It is wrong to include `:erlang.term_to_binary` in there:
- The payload is already a string (see the output of `event_to_payload/5)
- `term_to_binary` encodes a given term using the BEAM-specific encoding, that might differ from what one expects.


Indeed
```elixir
# An example of event_to_payload output
"{\"i_am\":\"a_map\"}" |> byte_size()
# > 16
"{\"i_am\":\"a_map\"}" |> :erlang.term_to_binary() |> byte_size()
# > 22
```